tidy3d/components/simulation.py

```diff
@@ -2217,6 +2337,68 @@ def _get_structures_plane(
                     medium_shapes.append((structure.medium, shape))
         return medium_shapes
 
+    def _get_structures_2dbox(
+        self,
+        structures: List[Structure],
+        x: float = None,
+        y: float = None,
+        z: float = None,
+        hlim: Tuple[float, float] = None,
+        vlim: Tuple[float, float] = None,
+    ) -> List[Tuple[Medium, Shapely]]:
+        """Compute list of shapes to plot on 2d box specified by (x_min, x_max), (y_min, y_max).
+
+        Parameters
+        ----------
+        structures : List[:class:`.Structure`]
+            list of structures to filter on the plane.
+        x : float = None
+            position of plane in x direction, only one of x, y, z must be specified to define plane.
+        y : float = None
+            position of plane in y direction, only one of x, y, z must be specified to define plane.
+        z : float = None
+            position of plane in z direction, only one of x, y, z must be specified to define plane.
+        hlim : Tuple[float,float] = None
+            Horizontal bounds for finding structures
```

Comment:
probably we can use the same discussion from the `.plot()` docstring here about which dimensions these refer to given what plane we are looking at.